### DISCUSSION TITLE
How to pass an argument with spaces to exec?

### DISCUSSION THREAD

**Discussion by sffc** [sagiegurari/duckscript#283] [Q&A]
I'm trying to write the following Bash command in Duckscript:

```bash
$ gn gen --root=../.. --args="target_os=\"wasi\" target_cpu=\"wasm32\"" out/wasi
```

However, it seems that the following doesn't work.

```duckscript
exec --fail-on-error gn gen --root=../.. --args="target_os=\"wasi\" target_cpu=\"wasm32\"" out/wasi
```

From a bit of debugging, it appears that the long string argument is getting tokenized because it has a space, despite being inside of quotation marks. I haven't figured out a way around this, especially because I have nested quotation marks.

**Comment by sagiegurari** [comment]
so many " inside " :)
maybe (not tested) something like...

```sh
args = set "target_os=\\\"wasi\\\" target_cpu=\\\"wasm32\\\""
exec --fail-on-error gn gen --root=../.. --args=\"${args}\" out/wasi
```

**Reply by sffc** [reply]
Hmm, if I do this, I get a different error:

```
ERROR at the command-line "--args":1:1: Expecting assignment or function call.
"target_os=\"wasi\" target_cpu=\"wasm32\""
^-----------------------------------------
```
